### DISCUSSION TITLE
Upgrade Mastodon v4.2.0-beta3

### DISCUSSION THREAD

**Discussion by DuSisen** [mastodon/mastodon#26818] [Q&A]
I installed Mastodon (top-netz.de) on a specially newly installed Debian 12 server using the Mastodon installation guide (https://docs.joinmastodon.org/admin/install/), put it into operation and thus strengthened the Mastodon network.
Now my concern: the version installed according to the instructions is: Mastodon v4.2.0-beta2. I get the information that Mastodon v4.2.0-beta3 is currently available and I can't find any description of how to specifically perform this update from Mastodon v4.2.0-beta2 to Mastodon v4.2.0-beta3. Isn't there a description for this? Or was I just too ignorant to find the right description? Or should I wait until the final version is upgradeable?

Original:
Ich habe auf einem speziell neu istallierten Debian 12 Server Mastodon (top-netz.de) mit Hilfe der Mastodon Installationsanleitung (https://docs.joinmastodon.org/admin/install/) installiert, in Betrieb genommen und damit das Mastodon-Netzwerk verstärkt.
Nun mein Anliegen: die laut Anleitung installierte Version ist: Mastodon v4.2.0-beta2. Ich erhalte die Information, dass aktuell Mastodon v4.2.0-beta3 vorhanden ist und finde keine Beschreibung, wie ich speziell diese Aktualisierung von Mastodon v4.2.0-beta2 zu Mastodon v4.2.0-beta3 durchführen kann. Gibt es dafür keine Beschreibung? Oder war ich lediglich zu unwissend die richtige Beschreibung zu finden? Oder sollte ich warten, bis die finale Version upgradebar ist?

**Comment by p37307** [comment]
Check the **Upgrade Notes** section in the release page.
 https://github.com/mastodon/mastodon/releases/tag/v4.2.0-beta3

**Reply by DuSisen** [reply]
Thanks for the tip, I had to do the following:
```
su - mastodon
cd /home/mastodon/live
git fetch && git checkout v4.2.0-beta3
exit
bundle config deployment 'true'
bundle config without 'development test'
bundle install -j$(getconf _NPROCESSORS_ONLN)
systemctl daemon-reload
systemctl enable --now mastodon-web mastodon-sidekiq mastodon-streaming
```
Then show me Mastodon v4.2.0-beta3 on the home page [https://top-netz.de](https://top-netz.de)

**Comment by DuSisen** [comment]
FYI, after upgrading to Mastodon v4.2.0-beta3 yesterday, I read the following error in the postgresql-15-main.log log:
`2023-09-06 23:57:59.349 CEST [463659] mastodon@mastodon_production FEHLER:  Relation »software_updates« existiert nicht bei Zeichen 34`
`2023-09-06 23:57:59.349 CEST [463659] mastodon@mastodon_production ANWEISUNG:  SELECT "software_updates".* FROM "software_updates" ORDER BY "software_updates"."id" ASC LIMIT $1`
und:
`2023-09-06 23:35:19.549 CEST [463664] mastodon@mastodon_production FEHLER:  doppelter Schlüsselwert verletzt Unique-Constraint »index_status_stats_on_status_id«`
`2023-09-06 23:35:19.549 CEST [463664] mastodon@mastodon_production DETAIL:  Schlüssel »(status_id)=(110650946685090411)« existiert bereits.`
Note: I have set my primary system language to German so I can read faster at log file level.
Addendum: These errors are not noticeable in the web interface.
